### PR TITLE
Buttons can now be held down

### PR DESCRIPTION
**Pull request by xathon**
Hold down appropiate buttons (Left/Right/Up/Down, Lens Shift, Focus, Zoom) to repeat the API calls 3 times per second (this number can be changed by editing one js var)

### CODE REVIEW

**Comment by localleon** [comment]
Hello @xathon ,

thanks for your PR. This is a really nice idea and I would love to merge it.

However, there's one problem. It's really difficult to review the code because you changed every single line in the file. This sometimes happens if you copy paste files or apply syntax linting. Now, it's not possible to see which lines were changed with this commit, which defeats the purpose of using Git overall.

Please update your PR/Branch with **a commit that only changes the lines which you changed**. Then I will gratefully review your PR.

PS: Your commit should look something like [this](https://github.com/localleon/barco-slm-network/commit/a157f674d923da62e0a10aa28fdbbb65980c6a68)

File: go/static/index.js [hunk 1]
```diff
@@ -1,161 +1,217 @@
...
+        default:
+        //alert(key); //Everything else
+    }
+
+}
+
+var tid; //timeout ID
+var persec = 3; //How many times per second should the API be called?
+
+function mdown(cmd) {
```
[the remark this answers is not in the document]
Comment (localleon) [reply go/static/index.js]: We could also implement the parameters `cmd, opt` directly like in `apirequest(cmd,opt)`. Then we wouldn't need that long `switch case`

Something like this (pseudocode):
```
function mdown(cmd, opt){
      apirequest(cmd,opt);
      tid = setTimeout(mdown.bind(null, cmd), Math.round(1000 / persec)); //Repeats the function with the same parameter n times per second, changable via persec variable
}
```
Is this a viable approach? WDYT?

File: go/static/index.js [hunk 2]
```diff
@@ -1,161 +1,217 @@
...
+        case 'shright':
+            apirequest('lensshift', 'right');
+            break; //shift lens to the right
+
+    }
+
+
+    tid = setTimeout(mdown.bind(null, cmd), Math.round(1000 / persec)); //Repeats the function with the same parameter n times per second, changable via persec variable
+}
+function mup() {
```
Comment (localleon) [inline go/static/index.js]: Maybe rename this as well? We should think of a schema for these functions!

File: go/static/index.js [hunk 3]
```diff
@@ -1,161 +1,217 @@
...
+            apirequest('infrared', 'exit');
+            break; //esc key
+        default:
+        //alert(key); //Everything else
+    }
+
+}
+
+var tid; //timeout ID
+var persec = 3; //How many times per second should the API be called?
```
Comment (localleon) [inline go/static/index.js]: Does this need to be global?

Comment (xathon) [reply go/static/index.js]: No, it doesn't. The global aspect is leftover from a different approach I had, but didn't use. I'll change that and put it in the function.

File: go/static/index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (xathon) [reply go/static/index.js]: > We could also implement the parameters cmd, opt directly like in apirequest(cmd,opt).

That makes a lot of sense, it should also make the code a lot easier to read.

>This function name is not very self-explaining. Maybe we could name it something like apiLongDown

I'd propose `apiButtonHold` and `apiButtonRelease`. That way, the relations to the API, the on-screen buttons and the "holding down" are obvious.

[the remark this answers is not in the document]
Comment (localleon) [reply go/static/index.js]: Perfect, it would be great if you could implement this!

File: go/static/index.js (hunk 2 above)
Comment (localleon) [reply go/static/index.js]: We found a naming schema in the question above.

File: go/static/index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (localleon) [reply go/static/index.js]: Latest commit works!

**Comment by localleon** [comment]
Thanks for your contribution! You rock!

**Comment by localleon** [comment]
Duplicate of #11

**Pull request closed without merging**